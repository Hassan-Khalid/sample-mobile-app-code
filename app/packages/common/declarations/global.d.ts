 export interface currentUser {
  __typename: string;
  _id: string;
  access: boolean;
  boards: any[];
  callChannels: string[];
  chatKeys: {
    publishKey: string;
    subscribeKey: string;
  };
  chatToken: string;
  chatUUID: string;
  email: string;
  emailNotification: boolean;
  fileImporter: {
    __typename: string;
    _id: string;
    importers: any[];
    licenseKey: string;
    user: {
      companyId: string;
      companyName: string;
      email: string;
      id: string;
      name: string;
    };
  };
  foldersAccessToken: string;
  info: string;
  intercomHash: string;
  isOem: boolean;
  name: string;
  notification: {
    __typename: string;
    email: {
      __typename: string;
      onAddedAsWorkOrderFollower: boolean;
      onAssignedTicketInternalNotePost: boolean;
      onAssignedWorkOrderUpdate: boolean;
      onMentionedInWorkOrderInternalNote: boolean;
      onMessageOnAssignedWorkOrder: boolean;
      onNewTicketAssigned: boolean;
    };
  };
  notificationChannel: string;
  notificationChannelGroupName: string;
  oem: {
    __typename: string;
    _id: string;
    admin: {
      __typename: string;
      _id: string;
    };
    allowFollowersMyWorkOrders: boolean;
    emailAccountLinked: boolean;
    emailAccountStatus: string | null;
    linkedEmailAddress: string | null;
    name: string;
    paidFeatures: string[];
    slug: string;
    statuses: any[][];
    urlOem: string;
  };
  role: string;
  teams: any[];
  userCredentialsSent: boolean;
}

export interface WorkOrderListTypeObject {
   __typename: string;
   _id: string;
   assignee: {
     __typename: string;
     _id: string;
     name: string;
   };
   createdAt: string;
   customFields: any[]; // You might want to specify the type for customFields
   facility: {
     __typename: string;
     _id: string;
     customFields: any[]; // You might want to specify the type for customFields
     name: string;
   };
   followers: any[]; // You might want to specify the type for followers
   hasOriginalEventBeenDeleted: null | boolean;
   machine: {
     __typename: string;
     _id: string;
     name: string;
     serialNumber: string;
   };
   maintenanceOn: string | null; // You might want to specify the type for maintenanceOn
   procedures: any[]; // You might want to specify the type for procedures
   schedule: string | null; // You might want to specify the type for schedule
   status: string;
   ticketChatChannels: string[]; // You might want to specify the type for ticketChatChannels
   ticketId: string;
   ticketInternalNotesChatChannels: string[]; // You might want to specify the type for ticketInternalNotesChatChannels
   ticketType:{
     _id: string
     name:string
     icon:string
     color:string
   };
   title: string;
   unread: boolean;
   updatedAt: string;
   user: any; // You might want to specify the type for user
 }
 
 type workOrderTicketObjectType = {
   __typename: string;
   _id: string;
   assignee: {
     __typename: string;
     _id: string;
     access: boolean;
     callChannels: string[];
     chatKeys: {
       publishKey: string;
       subscribeKey: string;
     };
     chatToken: string | null;
     chatUUID: string;
     department: string | null;
     email: string;
     emailNotification: boolean;
     firstName: string | null;
     function: string | null;
     info: string;
     isOem: boolean;
     lastName: string | null;
     mobile: string | null;
     name: string;
     notification: {
       __typename: string;
       email: any; // You need to define the structure for this
     };
     phone: string | null;
     role: string;
     teams: any[]; // You need to define the structure for this
     userCredentialsSent: boolean;
   };
   createdAt: string;
   customFields: any[]; // You need to define the structure for this
   description: string;
   facility: {
     __typename: string;
     _id: string;
     createdAt: string;
     customFields: any[][]; // You need to define the structure for this
     facilityId: string;
     isQRCodeEnabled: boolean;
     name: string;
     teams: any[]; // You need to define the structure for this
     updatedAt: string;
   };
   followers: any[]; // You need to define the structure for this
   hasOriginalEventBeenDeleted: any;
   inventoryParts: any[]; // You need to define the structure for this
   lastUpdatedAt: string;
   machine: {
     __typename: string;
     _id: string;
     createdAt: string;
     customFields: any[][]; // You need to define the structure for this
     customer: {
       __typename: string;
       _id: string;
       facilityId: string;
       name: string;
     };
     description: string;
     documentFolders: {
       __typename: string;
       externalId: string;
       internalId: string;
     };
     folderId: string | null;
     generalAccessUrl: string;
     image: string;
     inventoryParts: any[]; // You need to define the structure for this
     isBoxFoldersDisabled: boolean;
     isQRCodeEnabled: boolean;
     name: string;
     oem: {
       __typename: string;
       _id: string;
       logo: string;
       name: string;
       statuses: any[]; // You need to define the structure for this
       thumbnail: string;
     };
     productionLine: any; // You need to define the structure for this
     serialNumber: string;
     slug: string;
     teams: any[]; // You need to define the structure for this
     template: any; // You need to define the structure for this
     thumbnail: string;
   };
   maintenanceOn: any;
   notes: any;
   procedures: any[]; // You need to define the structure for this
   schedule: any;
   status: string;
   teams: any[]; // You need to define the structure for this
   ticketChatChannels: string[];
   ticketId: string;
   ticketInternalNotesChatChannels: string[];
   ticketType: {
     __typename: string;
     _id: string;
     color: string;
     icon: string;
     name: string;
   };
   timeTrackerLogs: {
     __typename: string;
     _id: string;
     createdBy: any; // You need to define the structure for this
     description: string;
     endDateTime: string;
     isBillable: boolean;
     startDateTime: string;
     ticketTag: any; // You need to define the structure for this
     timeElapsedInSeconds: number;
   }[];
   title: string;
   totalTimeLoggedInSeconds: number;
   unread: boolean;
   updatedAt: string;
   url: string | null;
   user: {
     __typename: string;
     _id: string;
     access: boolean;
     callChannels: string[];
     chatKeys: {
       publishKey: string;
       subscribeKey: string;
     };
     chatToken: string | null;
     chatUUID: string;
     department: string | null;
     email: string;
     emailNotification: boolean;
     firstName: string | null;
     function: string | null;
     info: string;
     isOem: boolean;
     lastName: string | null;
     mobile: string | null;
     name: string;
     notification: {
       __typename: string;
       email: any; // You need to define the structure for this
     };
     phone: string | null;
     role: string;
     teams: any[]; // You need to define the structure for this
     userCredentialsSent: boolean;
   };
 };
