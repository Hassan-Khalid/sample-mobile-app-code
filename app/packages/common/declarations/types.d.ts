import { TextStyle, Viewstyle } from 'react-native';
import * as string_decoder from 'string_decoder';
export type DeviceOrientation = 'portrait' | 'landscape' | null;

// eslint-disable-next-line
type AnyType = any;

export interface CommonAppConstants {
  androidPlatform: string;
  iOsPlatform: string;
  DeviceOrientation: {
    portrait: DeviceOrientation;
    landscape: DeviceOrientation;
  };
  splashTimeout: number,
  tabs:{}
}
export type ApplicationColors = {
  GRAY1: string;
  GRAY1_DARK: string;
  GRAY1_LIGHT: string;
  GRAY1_IMG_TEXT: string;
  GRAY2: string;
  GRAY2_DARK: string;
  GRAY2_LIGHT: string;
  GRAY3: string;
  GRAY3_BACKGROUND: string;
  GRAY3_DARK: string;
  GRAY3_LIGHT: string;
  BACKGROUND: string;
  WHITE: string;
  WHITE_PERSIST: string;
  WARNING: string;
  WARNING_DARK: string;
  WARNING_LIGHT: string;
  DANGER: string;
  DANGER_DARK: string;
  DANGER_LIGHT: string;
  INFO: string;
  INFO_DARK: string;
  INFO_LIGHT: string;
  BLUE: string;
  BLACK: string;
  BLACK_PERSIST: string;
  SHADOW: string;
  APPROVED: string;
  DENIED: string;
  PENDING: string;
  IMG_TEXT: string;
  STATUS_1:string;
  STATUS_1_TEXT:string;
  BLUE_LIGHT:string
};

export interface InputProperties {
  input: EditableInput;
  CustomInput?: React.ForwardRefExoticComponent<
    {customFieldControl: CustomFieldControl| undefined } & React.RefAttributes<unknown>
  >;
  containerHandleBlur?: (value?: string) => void;
  containerHandleOnPress?: Function | undefined;
  containerHandleChange?: Function | undefined;
}
export interface ThemedButtonProps {
  title: string;
  onPress: () => void;
  buttonStyle?: Viewstyle;
  titleStyle?: TextStyle;
  disabled?: boolean;
  loading?: boolean;
  size?: 'large' | 'medium' | 'small';
  type?: 'solid' | 'clear' | 'outline';
}
export type buttonType = {
  title?:string;
  titleStyle?:TextStyle;
  onPress:()=>void;
  buttonStyle?: Viewstyle;
  icon?: AnyType
}

export interface ScreenHeader {
  componentId: string;
  leftButtons?:buttonType ;
  title?: string | undefined;
  rightButtons?: buttonType;
  getSearchText?:(text:string)=>void
}

export interface ToastProps {
  message: string;
  type: 'normal' | 'success' | 'warning' | 'danger' | 'custom',
  placement?: 'top' | 'bottom',
  duration?: number,
  animationType?: 'slide-in' | 'zoom-in'
}

interface ILandingService {
  // getAppConfig<T>(apiPrefix: string): Promise<ServiceResponse<T>>;
  
  getUser(): AnyType;
  
  // getAppConfigApi(): string;
  
  // getApplicationId(): number;
}

interface IWorkOrderListService {
  getMyWorkOrderList():AnyType
  getWorkOrderList():AnyType;
}

interface IWorkOrderDetailService {
  getWorkOrderByID(id:string):AnyType
}

export type Status = {
  __typename: string;
  _id: string;
  color: string;
  label: string;
};