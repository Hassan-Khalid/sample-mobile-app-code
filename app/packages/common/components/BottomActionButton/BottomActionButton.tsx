import React from "react"
import { FlatList, View } from 'react-native';
import {ThemedButtonProps} from "../../declarations/types"
import ThemedButton  from '../ThemedButton/ThemedButton';
import styles from './styles/BottomActionButton.style';

type BottomActionButtonProps = {
  buttonArray:ThemedButtonProps[]
}

const BottomActionButton =  ({ buttonArray }:BottomActionButtonProps) => {
  return (
    <View style={{...styles.container,...styles.shadowContainer}}>
      <FlatList
        scrollEnabled={false}
        keyExtractor={(item, index) => `${index}`}
        contentContainerStyle={{...styles.buttonContainer}}
        data={buttonArray}
        renderItem={ ({ item })=> <ThemedButton {...item}/>}
      />
    </View>
  )
  
}

export default  BottomActionButton