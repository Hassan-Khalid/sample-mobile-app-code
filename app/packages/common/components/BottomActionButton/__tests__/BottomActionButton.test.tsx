import React from 'react';
import { render } from '@testing-library/react-native';
import BottomActionButton from '../BottomActionButton';
import { ThemedButtonProps } from 'common/declarations/types';

describe('BottomActionButton', () => {
  it('renders correctly', () => {
    // exercise
    const buttonArray:ThemedButtonProps   = {title:'Login',onPress:()=>{},size:'large',type:'solid'};
    
    const wrapper = render(<BottomActionButton buttonArray={[buttonArray]} />);
    
    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
  it('should render empty', () => {
    // exercise
    const loading = false;
    const wrapper = render(<BottomActionButton buttonArray={[]} />);
    
    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
