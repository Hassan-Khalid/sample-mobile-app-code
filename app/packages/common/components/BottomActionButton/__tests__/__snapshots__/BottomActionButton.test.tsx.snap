// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomActionButton renders correctly 1`] = `
<View
  style={
    {
      "ZIndex": 9999,
      "backgroundColor": {
        "dynamic": {
          "dark": "#FFF",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#FFFFFF",
        },
      },
      "borderTopColor": {
        "dynamic": {
          "dark": "#77889d",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#77889d",
        },
      },
      "borderTopWidth": 0,
      "borderWidth": 1,
      "bottom": 1,
      "height": 80,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 5,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
        "backgroundColor": {
          "dynamic": {
            "dark": "#FFF",
            "highContrastDark": undefined,
            "highContrastLight": undefined,
            "light": "#FFFFFF",
          },
        },
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "space-evenly",
      }
    }
    data={
      [
        {
          "onPress": [Function],
          "size": "large",
          "title": "Login",
          "type": "solid",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 2,
              },
              undefined,
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityLabel="button-component"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="button-component"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": {
                    "dynamic": {
                      "dark": "#64b4fa",
                      "highContrastDark": undefined,
                      "highContrastLight": undefined,
                      "light": "#0000EE",
                    },
                  },
                  "borderColor": "#2089dc",
                  "borderRadius": 10,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "marginHorizontal": 30,
                  "padding": 8,
                  "paddingHorizontal": 10,
                  "width": 700,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`BottomActionButton should render empty 1`] = `
<View
  style={
    {
      "ZIndex": 9999,
      "backgroundColor": {
        "dynamic": {
          "dark": "#FFF",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#FFFFFF",
        },
      },
      "borderTopColor": {
        "dynamic": {
          "dark": "#77889d",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#77889d",
        },
      },
      "borderTopWidth": 0,
      "borderWidth": 1,
      "bottom": 1,
      "height": 80,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 5,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
        "backgroundColor": {
          "dynamic": {
            "dark": "#FFF",
            "highContrastDark": undefined,
            "highContrastLight": undefined,
            "light": "#FFFFFF",
          },
        },
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "space-evenly",
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
